### ISSUE TITLE
Accessors declared as immutable collections/maps should generate same builder code as non-immutable

### ISSUE THREAD

**Issue by tbroyer** [immutables/immutables#86]
For example, if I use:

```java
@Value.Immutable
interface Person {
  List<PhoneNumber> phoneNumbers();
}
```

the builder will internally use an `ImmutableList.Builder` and will have `addPhoneNumber` and `addAllPhoneNumbers`.

But if I use `ImmutableList` instead of `List` (because I want to be explicit that the list is immutable, and possibly avoid some calls to `ImmutableList.copyOf()` when some API expects an `ImmutableList` as input):

```java
@Value.Immutable
interface Person {
  ImmutableList<PhoneNumber> phoneNumbers();
}
```

then the generated builder will use an `ImmutableList` with only a `setPhoneNumbers`. IMO, it should use an `ImmutableList.Builder` as in the first case.

**Comment by elucash** [comment]
It was one of the early design decisions to do it as it is. Now I'm thinking the opposite, why not make immutable collections to be handled the same way? There are other means to turn it to regular attribute if needed (for example make it `@Nullable`).
Now I strongly think that we should support this. Other parties will also benefit from it. Facebook Buck team also asked for this, or more precisely: was wondering why it haven't been done in a way to support `ImmutableList` etc.
Thank you for proposal!

**Comment by elucash** [comment]
Released in 2.0
